icechunk-python/src/store.rs

```diff
@@ -76,58 +76,48 @@ impl PyStore {
                 e
             ))
         })?;
-        Ok(Self(Arc::new(RwLock::new(store))))
+        Ok(Self(Arc::new(store)))
     }
 
-    fn __eq__(&self, other: &Self) -> PyIcechunkStoreResult<bool> {
-        pyo3_async_runtimes::tokio::get_runtime().block_on(async move {
-            let session_id = self.0.read().await.session_id().await?;
-            let other_session_id = other.0.read().await.session_id().await?;
-            Ok(session_id == other_session_id)
-        })
-    }
+    fn __eq__(&self, other: &PyStore) -> bool {
+        if Arc::ptr_eq(&self.0, &other.0) {
+            return true;
+        }
 
-    #[getter]
-    fn session_id(&self) -> PyIcechunkStoreResult<String> {
-        pyo3_async_runtimes::tokio::get_runtime().block_on(async move {
-            let session_id = self.0.read().await.session_id().await?;
-            Ok(session_id.to_string())
-        })
+        // If the stores were created from the same session they are equal
+        let session = self.0.session();
+        let other_session = other.0.session();
+        Arc::ptr_eq(&session, &other_session)
```

Comment:
I think you don't need the previous if? This second condition will also match in the first case

Reply:
Ah yeah that's a relic from before I got rid of the inner lock I didn't need